test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -1013,7 +1014,19 @@ public void testSupportedFieldTypes() throws IOException {
                     // TODO in the future we can make this more explicit with expectThrows(), when the exceptions are standardized
                     AssertionError failure = null;
                     try {
-                        searchAndReduce(indexSearcher, new MatchAllDocsQuery(), aggregationBuilder, fieldType);
+                        InternalAggregation internalAggregation = searchAndReduce(
+                            indexSearcher,
+                            new MatchAllDocsQuery(),
+                            aggregationBuilder,
+                            fieldType
+                        );
+                        // We should make sure if the builder says it supports sampling, that the internal aggregations returned override
+                        // finalizeSampling
+                        if (aggregationBuilder.supportsSampling()) {
```

Comment:
@not-napoleon I am not 100% where to put this test. We really do need a test that builds a builder, and then creates the internal agg representation to verify that `finalizeSampling` is satisfied. It fits perfectly here. 

Another option is to effectively copy-paste this whole test case and instead of checking types we run this check. But that felt wasteful 🤷